docs/naps/0-nap-process.md

```diff
@@ -0,0 +1,298 @@
+applicable.
+
+We intend NAPs to be the primary mechanisms for proposing major new
+features, for collecting community input on an issue, and for documenting
+the design decisions that have gone into napari. The NAP author is
+responsible for building consensus within the community and documenting
+dissenting opinions.
+
+Because the NAPs are maintained as text files in a versioned repository,
+their revision history is the historical record of the feature proposal
+[^id3].
+
+### When is a NAP warranted?
+
+Most contributions to napari should go through the standard [contributing
+process](napari-contributing), that is, opening a pull request to the main
+repository. They will typically be uncontroversial improvements, and
+require little design discussion. The git commit and the pull request
+itself serve as an adequate record of the contribution's history.
+
+In some cases, contributions will require extensive discussions around any
+new APIs, breaking of existing APIs, changing of governance, build, or
+contributing mechanisms, or other aspects of the project. These might
+happen both on the PR itself and on other channels, such as Zulip,
+community meetings, or even one on one discussions. In such situations, the
+PR will not contain sufficient information to document all the
+considerations that went into a decision. Core developers may at their
+discretion then call for a NAP to summarize the discussion to date.
+
+In addition to the above situation, napari is the product of many
+historical decisions that were *not* explained by a NAP. In situations
+where community members might be confused about parts of napari's design,
+and whether alternate designs were considered and rejected, or simply not
+considered. It might then be useful to write a retrospective informational
+NAP to explain that aspect of the project.
+
+### Types of NAPs
+
+There are three kinds of NAPs:
+
+1. A **Standards Track** NAP describes a new feature or implementation
+   for napari.
+2. An **Informational** NAP describes a napari design issue, or provides
+   general guidelines or information to the Python community, but does not
+   propose a new feature. Informational NAPs do not necessarily represent a
+   napari community consensus or recommendation, so users and
+   implementers are free to ignore Informational NAPs.
+3. A **Process** NAP describes a process surrounding napari, or
+   proposes a change to (or an event in) a process. Process NAPs are
+   like Standards Track NAPs but apply to areas other than the napari
+   library itself. They may propose an implementation, but not to
+   napari's codebase; they require community consensus. Examples include
+   procedures, guidelines, changes to the decision-making process, and
+   changes to the tools or environment used in scikit-image development.
+   Any meta-NAP is also considered a Process NAP.
+
+## NAP Workflow
+
+The NAP process begins with a new idea for napari. A NAP should contain a
```

Comment:
I'm curious about the nature of this "newness"- how you see NAPs relating to current/future roadmaps (ideally we get better at sharing these publically). It'd be great to know if a NAP tags to a priority in an existing roadmap, pushes thoughtfully against an existing priority (and the argument is therefore it should get implemented sooner rather than later), or is a mechanism to introduce a future idea for a down the road implementation? For design and UXR, this would be helpful to flag and assess (getting ahead of the new ideas to help support them with interviews, clarity mapping, etc, is a value add UXR can support).

Reply:
I would say that prioritisation in a broad community project like napari happens outside of NAPs. A NAP is a method to decide *whether or not* a new idea will make it into napari. It is up to individual contributors to implement that idea, and nothing we write on a NAP can make that happen.

As regards roadmaps, ideally, everything on a roadmap has a corresponding Provisional or Accepted NAP, but it will take some time to get to that state, since NAPs are not yet implemented and there's lots of ideas we've agreed on already.

Also, in some cases a roadmap might say "we are prioritising drawing down bugs this next quarter", and one certainly does not need a NAP to fix a bug. :stuck_out_tongue_winking_eye: